mmv1/third_party/terraform/services/resourcemanager/resource_google_service_account.go

```diff
@@ -30,6 +31,7 @@ func ResourceGoogleServiceAccount() *schema.Resource {
 		},
 		CustomizeDiff: customdiff.All(
 			tpgresource.DefaultProviderProject,
+			resourceGoogleServiceAccountPlanOutputs,
```

Comment:
Sure, we often find ourselves in a situation where it makes sense to use `for_each` to assign various IAM roles or various other tasks referencing multiple service accounts by their email/member (e.g. old-style Workload Identity Federation), but because those fields are not available at apply time you get this error:
```
│ Error: Invalid for_each argument
│ 
│   on .terraform/modules/my_module_name/main.tf line 8, in resource "google_cloud_identity_group_membership" "this":
│    8:   for_each = toset(var.service_accounts)
│     ├────────────────
│     │ var.service_accounts is list of string with 1 element
│ 
│ The "for_each" set includes values derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will identify the instances of this resource.
```
We typically end up templating the email string as a workaround, but if the SA is already created and someone adds a reference using these fields there's no obvious problem when it's applied. You won't find out until the next time that TF is run from a clean slate however, such as in a DR scenario

Reply:
When trying to reproduce this, I ran into the following (slightly different) error message:

```
│ Error: Invalid for_each argument
│ 
│   on pr-11919.tf line 10, in resource "google_project_iam_member" "iam_members":
│   10:   for_each = toset([google_service_account.service_account.email, google_service_account.service_account_2.email])
│     ├────────────────
│     │ google_service_account.service_account.email is a string, known only after apply
│     │ google_service_account.service_account_2.email is a string, known only after apply
│ 
│ The "for_each" set includes values derived from resource attributes that
│ cannot be determined until apply, and so Terraform cannot determine the full
│ set of keys that will identify the instances of this resource.
│ 
│ When working with unknown values in for_each, it's better to use a map value
│ where the keys are defined statically in your configuration and where only
│ the values contain apply-time results.
│ 
│ Alternatively, you could use the -target planning option to first apply only
│ the resources that the for_each value depends on, and then apply a second
│ time to fully converge.
```

The `-target` option doesn't really seem like a great solution, though. I assume you're doing something similar but storing it in a var somehow, resulting in a different error message.